File: cpp/Car.cpp
#include "Car.hpp"

// Constructor
Car::Car(const std::string &plate, const std::string &brand, int year, const std::string &clr,
        int capacity, int doors)
    : Vehicle(plate, brand, clr, year), humanCapacity(capacity), numberOfDoors(doors) {
}

// Destructor
Car::~Car() {
}

// Getters
int Car::getHumanCapacity() const {
    return humanCapacity;
}

int Car::getNumberOfDoors() const {
    return numberOfDoors;
}

// Setters
void Car::setHumanCapacity(int capacity) {
    humanCapacity = capacity;
}

void Car::setNumberOfDoors(int doors) {
    numberOfDoors = doors;
}

File: python/Car.py
from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, plate, brand, year, clr, capacity, doors):
        super().__init__(plate, brand, clr, year)
        self._humanCapacity = capacity
        self._numberOfDoors = doors

    # Getters
    def getHumanCapacity(self):
        return self._humanCapacity

    def getNumberOfDoors(self):
        return self._numberOfDoors

    # Setters
    def setHumanCapacity(self, capacity):
        self._humanCapacity = capacity

    def setNumberOfDoors(self, doors):
        self._numberOfDoors = doors
